var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -414,3 +572,232 @@ def cmake_args(self):
             args.append("-DCMAKE_CXX_COMPILER_FORCED=True")
 
         return args
+
+
+class MakefileBuilder(spack.build_systems.makefile.MakefileBuilder):
+    build_directory = "src/fortran"
+
+    # Generate Compiler Specific includes
+    def edit(self, pkg, spec, prefix):
+        config = {
+            "FC": pkg.compiler.fc_names[0],
+            "FCFLAGS": "",
+            "ARCH": platform.machine(),
+            "DOCONCURRENT_FLAG": "",
+            "ARRAY_FLAG": "",
+            "OPENMP_FLAG": "",
+            "OPENACC_FLAG": "",
+            "CUDA_FLAG": "",
+            "SEQUENTIAL_FLAG": "",
+        }
+
+        # ===================================
+        #               ARM
+        # ===================================
+        if spec.compiler.name == "arm":
+            flags = "-std=f2018 " + pkg.compiler.opt_flags[4] + " -Wall -Wno-unused-variable"
+
+            if platform.machine() == "aarch64":
+                # TODO: Add MCPU option here as in -mcpu=$(MCPU)
+                flags += "-mcpu=native"
+            else:
+                # TODO: Add MARCH option here as in -march=$(MARCH)
+                flags += "-march=native"
+
+            config["DOCONCURRENT_FLAG"] = pkg.compiler.openmp_flag  # libomp.so required
+            config["ARRAY_FLAG"] = pkg.compiler.openmp_flag  # libomp.so required
+            config["OPENMP_FLAG"] = pkg.compiler.openmp_flag  # libomp.so required
+            config["OPENACC_FLAG"] = "-fopenacc"
+
+            # Error Messages
+            if self.spec.variants["foption"].value in [
+                "CUDA",
+                "CUDAKernels",
+                "OpenACC",
+                "OpenACCArray",
+            ]:
+                sys.exit(self.spec.variants["foption"].value + "is not supported by this compiler")
```

Comment:
Probably better to raise an error than to call `sys.exit` directly. Looks like most packages are using `InstallError` for similar issues.

Reply:
Thank you! I tried to also simplify checking process which reduced number of lines too